crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -4,21 +4,34 @@ use dashmap::DashMap;
 use reth_interfaces::RethResult;
 use reth_nippy_jar::NippyJar;
 use reth_primitives::{
-    snapshot::BLOCKS_PER_SNAPSHOT, Address, BlockHash, BlockHashOrNumber, BlockNumber, ChainInfo,
-    Header, SealedHeader, SnapshotSegment, TransactionMeta, TransactionSigned,
-    TransactionSignedNoHash, TxHash, TxNumber, B256, U256,
+    snapshot::{HighestSnapshots, BLOCKS_PER_SNAPSHOT},
+    Address, BlockHash, BlockHashOrNumber, BlockNumber, ChainInfo, Header, SealedHeader,
+    SnapshotSegment, TransactionMeta, TransactionSigned, TransactionSignedNoHash, TxHash, TxNumber,
+    B256, U256,
 };
 use std::{ops::RangeBounds, path::PathBuf};
+use tokio::sync::watch;
 
 /// SnapshotProvider
 #[derive(Debug, Default)]
 pub struct SnapshotProvider {
     /// Maintains a map which allows for concurrent access to different `NippyJars`, over different
     /// segments and ranges.
     map: DashMap<(BlockNumber, SnapshotSegment), LoadedJar>,
+    /// Tracks the latest and highest snapshot of every segment.
```

Comment:
I think only `highest` is correct here
```suggestion
    /// Tracks the highest snapshot of every segment.
```